HHS/Head-Start-TTADP
# [TTAHUB-374] save progress to local storage when creating an AR

## Description of change
When creating an activity report, data is saved to local storage. If a report is not retrieved or saved from the backend, you can proceed through the report as normal. There are messages that attempt to explain this to the user. In addition, submit report behavior has been modified so that if the backend fails to save or retrieve a report, submission is disabled until such time as the report can be retrieved.

In my conception, this is meant to 1) mitigate some issues with low/no internet and 2) provide peace of mind to our users that there is an additional layer of saving happening, which is **somewhat** protected against any weird errors they might see in the UI.

I also spent a while in the unit tests, so you'll see a few changes related to cleaning up the console.

Before deployment, the language on these error messages will have to be workshopped amongst the larger group since this is basically whatever fell out of my fingers at the time.

## How to test
Create a report as normal. It should work! If you check while creating a report in your local storage, you should see that the report has been "backed up" there. It is cleaned up once its in submitted or approved state (there are certain browsers and environments that limit the amount of local storage space, so cleanup is necessary).

For additional testing, you can add errors and return statements to the appropriate fetchers (getReport and saveReport) and create a report (the report will need to have an id first, so probably save one and then add the errors, and then continue working). You should be able to, and if you reconnect it should be fairly seemless. 

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-374

## Checklists

### Every PR



### Production Deploy


### After merge/deploy
